resteasy-client-reactor-netty/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ReactorNettyClientHttpEngine.java

```diff
@@ -43,20 +44,45 @@
     private final HttpClient httpClient;
     private final ChannelGroup channelGroup;
     private final ConnectionProvider connectionProvider;
+    private final Optional<Duration> timeout;
 
     /**
      * Constructor for ReactorNettyClientHttpEngine
      *
      * @param httpClient The {@link HttpClient} instance to be used by this {@link AsyncClientHttpEngine}
      * @param channelGroup The {@link ChannelGroup} instance used by the provided {@link HttpClient}
      * @param connectionProvider The {@link ConnectionProvider} instance used to create the provided {@link HttpClient}
+     * @param timeout The {@link Optional<Duration>} instance used to configure timeout on response
      */
-    public ReactorNettyClientHttpEngine(final HttpClient httpClient,
+    ReactorNettyClientHttpEngine(final HttpClient httpClient,
                                         final ChannelGroup channelGroup,
-                                        final ConnectionProvider connectionProvider) {
+                                        final ConnectionProvider connectionProvider,
+                                        final Optional<Duration> timeout) {
```

Comment:
There are are already two constructors:

1. One that takes a `Duration`
2. One that does not take `Duration`

I find it confusing to have a third one that takes `Optional<Duration>`.  The above two already conveys that it is optional.  Can you please clarify why do we still need this constructor?

Reply:
This is a package private constructor.

Found this pattern convincing for constructing object and also, avoids unnecessary doc repetition between constructors.

Reply:
If created for convenience for using from other constructors, let's make it `private` then.  If this constructor would be used anywhere else other than this class itself, it gets confusing.